<a href="https://colab.research.google.com/github/EmamulHossen/K-Nearest-Neighbors-Assignment-7.1/blob/main/KNN_for_Regression(7_1).ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [ ]:
from google.colab import drive

In [ ]:
drive.mount("/content/drive")

Drive already mounted at /content/drive; to attempt to forcibly remount, call drive.mount("/content/drive", force_remount=True).


**Import Library**

In [ ]:
import pandas as pd

**Import Dataset**

In [ ]:
df=pd.read_csv("/content/drive/MyDrive/Data/KNN_dataset.csv")

In [ ]:
df

age  Income(K)
0   21         60
1   20         55
2   22         60
3   22         61
4   23         65
5   21         62
6   25         65
7   30         70
8   31         68

**Separate Feature(X) & Label(Y)**

In [ ]:
x=df.drop(["Income(K)"],axis=1)
y=df["Income(K)"]

In [ ]:
x

age
0   21
1   20
2   22
3   22
4   23
5   21
6   25
7   30
8   31

In [ ]:
y

0    60
1    55
2    60
3    61
4    65
5    62
6    65
7    70
8    68
Name: Income(K), dtype: int64

# **Apply KNN Regression Model**

In [ ]:
from sklearn.neighbors import KNeighborsRegressor

In [ ]:
model=KNeighborsRegressor(n_neighbors=3)

In [ ]:
model.fit(x,y)

KNeighborsRegressor(n_neighbors=3)

In [ ]:
y_prediction=model.predict([[22]])

/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but KNeighborsRegressor was fitted with feature names
  warnings.warn(


In [ ]:
y_prediction

array([60.33333333])

# **Apply Linear Regression**

In [ ]:
from sklearn.linear_model import LinearRegression

In [ ]:
LR_model=LinearRegression()

In [ ]:
LR_model.fit(x,y)

LinearRegression()

In [ ]:
y_predict=LR_model.predict([[22]])

/usr/local/lib/python3.10/dist-packages/sklearn/base.py:439: UserWarning: X does not have valid feature names, but LinearRegression was fitted with feature names
  warnings.warn(


In [ ]:
y_predict

array([60.97068966])

# Compare two Models

**The predictive outcome of Linear Regression is 60.97 on the other hand the outcome of KNN is 60.33, which is same as the manually calculated.For this reason,the KNN model is the best for this dataset.**